docs/architecture/api-packages.md

```diff
@@ -0,0 +1,156 @@
+      "displayName": "",
+      "documentationURL": "",
+      "supportURL": ""
+    },
+    "events": [
+      "707ea9da-0bf8-4dad-8145-a934ee1ebffb",
+      "6e5969c6-63fd-42c0-af17-72ed95c9dcd5"
+    ],
+    "apis": ["foo", "bar"]
+  }
+}
+```
+
+The root object key defines API Package name. In this example, `foo` is the API Package name.
+
+In order to allow validation of the label value, Compass creates `api-packages` LabelDefinition automatically. It is done while creating new Application. Once Tenant management feature is ready, creating new tenant will be used as a trigger for the LabelDefinition creation.
+The `api-packages` LabelDefinition has the following JSON schema:
+
+```json
+{
+  "$id": "https://example.com/arrays.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "A representation of a person, company, organization, or place",
+  "type": "object",
+  "propertyNames": {
+    "pattern": "^[a-z0-9_-]*$",
+    "minLength": 3,
+    "maxLength": 35
+  },
+  "additionalProperties": {
+    "anyOf": [{ "$ref": "#/definitions/apiPackage" }]
+  },
+  "definitions": {
+    "apiPackage": {
+      "type": "object",
+      "required": ["events", "apis"],
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Description"
+        },
+        "metadata": {
+          "type": "object",
+          "description": "Additional Metadata for Service Class representation"
+        },
+        "events": {
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "pattern": "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+In a result, the `api-packages` label value is validated. Also, the API Package name uniqueness within Application is guaranteed. Event and API Definition existence is not checked, apart from verifying the UUID type of the ID arrays.
+
+### Assumptions
+
+- API Package name doesn't have to be unique within tenant.
+- One API or Event Definition can belong to only one API Package. If an API or Event Definition is assigned to multiple API packages, or some items with provided IDs doesn't exist, then Runtime Agent reports a configuration sync error.
+
+### Pros
+
+- No breaking changes in our API
+- Consistent implementation (identical to Scenarios)
+- Static label validation
+
+### Cons
+
+- Additional logic on client side (see the [Client Usage](#client-usage) section)
+- _(not important)_ No validation for existence of API/Event Definition IDs
+
+## Example
+
+> **NOTE:** For clarity, in this example IDs has been changed from UUID type to string.
+
+1. User creates Application `sample-app`.
+1. User registers three different API definitions with the following IDs: `foo`, `bar` and `baz`.
+1. In order to create API Package called `sample-package`, User sets the following label for `sample-app` Application:
+
+   ```yaml
+   api-packages:
```

Comment:
In the example above and in the JSON schema you operate on the JSON object where keys are the names of the api-packages. Here, on the other hand, you present the array of objects and the name is included in the object.

Reply:
I forgot to update this section. Sorry